docs/CONTRIBUTING.md

```diff
@@ -3,199 +3,288 @@ id: contributing
+        - If applied, this commit will _commit title here_
+- body:
+    - Separate from subject with a blank line
+    - Wrap texts at 72 characters
+    - Explain _what_ and _why_, not _how_
+    - [GitHub flavored Markdown](https://github.github.com/gfm/) is ok to use
+    - Start with `BREAKING CHANGE: ` when you made significant change in the commit (see versioning section below).
+
+Example commit message:
 
-```sh
-git clone --recursive {YOUR_FORKED_REPOSITORY}
-cd {YOUR_FORKED_REPOSITORY}
-yarn install
-yarn run bootstrap
 ```
+test(textlint-formatter): check types while testing
+
+- Add strict type check option to `ts-node` to make sure future
+  interface changes will be took into account while running test.
+- Update test case for interface changes made at #430.
+
+Closes #448.
+```
+
+Please see [Commit Message Format](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular), [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/), and [Semantic Commit Messages](https://seesparkbox.com/foundry/semantic_commit_messages) for detail.
+
+##### Versioning
 
-### Testing
+We care version number while releasing packages to npm registry so you should not modify `version` field of `package.json`. For the record, we use [Semantic Versioning](https://semver.org/).
 
-This repository has three type tests.
-You should run **Unit Test** before submit to Pull Request.
+- Patch release (intended to not break your lint build)
+    - A bug fix to the CLI or core (including formatters)
+    - Improvements to documentation
+    - Non-user-facing changes such as refactoring
+    - Re-releasing after a failed release (i.e., publishing a release that doesn't work for anyone)
+- Minor release (might break your lint build)
+    - A new option
+    - An existing rule is deprecated
+    - A new CLI capability is created
+    - New public API are added (new classes, new methods, new arguments to existing methods, etc.)
+        - It might break type interface(`.d.ts`)
+    - A new formatter is created
+- Major release (break your lint build)
+    - A new option to an existing rule that results in ESLint reporting more errors by default
+    - An existing formatter is removed
+    - Part of the public API is removed or changed in an incompatible way
 
-- Unit Test – Run [packages/*](../packages/) tests
-- Example Test – Run [examples/*](../examples/) as test
-- Integration Test – Run [Integration test](../test/integration-test)
+### Running Tests
+
+We have four type of tests. You should run at least **unit test** or **documentation test** according to your type of changes before submitting a pull request.
+
+All tests should be run at the top directory of your fork.
 
 #### Unit Test
 
-Test all [packages/*](../packages/)'s tests.
+Run tests under [`packages/`](../packages):
```

Comment:
Should we use a URL instead of relative link.

Because, it broke a link in https://deploy-preview-495--textlint.netlify.com/docs/contributing.html

Reply:
Interesting as I think Docusaurus take care of these. Thank you for pointing out. Will see later today.